### PR TITLE
[RW-6097][RW-6130][risk=no] Documenting steps to upload 1kg genomes

### PR DESCRIPTION
**Pull request by ericsong**
Description:

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI + API server with `yarn test-local`
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: api/genomics/wdl/ReblockGVCF.wdl [hunk 1]
```diff
@@ -1,31 +1,33 @@
...
-  String gvcf_basename = basename(gvcf, ".g.vcf.gz")
+  String output_gcs_bucket_no_trailing_slash = sub(output_gcs_bucket, "/$", "")
 
-  call Reblock {
-    input:
-      gvcf = gvcf,
-      gvcf_index = gvcf_index,
-      output_vcf_filename = gvcf_basename + ".reblocked.g.vcf.gz",
-      docker_image = docker_image
+  scatter (gvcf in read_lines(gvcfs_list)) {
```
Comment (calbach) [inline api/genomics/wdl/ReblockGVCF.wdl:17]: Out of curiosity, is this throttled in some manner by Cromwell? Or will it just attempt to queue up these tasks on the underlying execution system?

**Review by calbach — COMMENTED** [review]
Thanks for the writeup, looks reasonable

File: api/genomics/wdl/ReblockGVCF.wdl (hunk 1 above)
Comment (ericsong) [reply api/genomics/wdl/ReblockGVCF.wdl:17]: I think it's the latter. From personal experience, cromwell seems to just spin them up ASAP.

File: api/genomics/docs/uploading_1kg_wgs_test_data.md [hunk 2]
```diff
@@ -19,14 +19,23 @@ were copied into `gs://all-of-us-workbench-test-genomics/1kg_gvcfs/`
...
+- ```bq query --max_rows=1000000 --format=csv --nouse_legacy_sql 'SELECT sample_name FROM `all-of-us-workbench-test.wgs_demo.metadata`' | sed -e 1d > vcf_sample_names.txt```
 - ```paste -d "," <(shuf -n $(cat vcf_sample_names.txt | wc -l) ehr_sample_names.txt) vcf_sample_names.txt > old_new_sample_names_map.csv```
 - ```cat old_new_sample_names_map.csv | awk -F',' '{print "UPDATE `all-of-us-workbench-test.wgs_demo_7.metadata` set sample_name=\""$1"\" where sample_name=\""$2"\";"}'```
 - Run update queries in BQ console or bq CLI
 
+### Extracting cohort
+- The first few runs of the extract failed until I tweaked some parameters
+- The scatter count was increased to 1000 to let the extraction finish in a reasonable time frame (~3 hours). 
+- The extraction time per shard varies a lot right now. Probably due to some more complex genomic intervals containing many more variants than less complex regions.
+- Some of the shards were also failing until I turned down the parameter for max local records in memory from 10 million to 5 million.
```
Comment (calbach) [inline api/genomics/docs/uploading_1kg_wgs_test_data.md:38]: Failing == OOM, right? I'm wondering whether we should considering having this parameter be a function of the number samples in the cohort being extracted for the service method. Other than that, I'm imagining the memory overhead of an average row should be uniform across cohorts and datasets.

**Review by calbach — APPROVED** [review]

File: api/genomics/docs/uploading_1kg_wgs_test_data.md (hunk 2 above)
Comment (ericsong) [reply api/genomics/docs/uploading_1kg_wgs_test_data.md:38]: I couldn't pinpoint is as OOM from the logs but the problem disappeared after changing this parameter so I think it's likely. We should adjust the parameter automatically once we figure out what the right coefficients are. I think the interval length and location probably also matter.

**Pull request merged**